pages/page/[page].tsx

```diff
@@ -25,25 +28,42 @@ export const getStaticProps: GetStaticProps = async (context) => {
   const parser = new Parser();
   let docs = cache.get<Doc[]>('docs');
   if (!docs) {
-    docs = (
-      await Promise.all(
-        channelsData.channels.map(async (channel, channelIndex) => {
-          const result = await parser.parseURL(channel.url);
-          return (
-            result?.items.map((item) => ({
-              ...item,
-              author: channel,
-              authorId: channelIndex
-            })) ?? []
-          );
-        })
-      )
-    ).flat();
-    docs.sort((a, b) => {
-      let da = new Date(b.pubDate as string);
-      let db = new Date(a.pubDate as string);
-      return +da - +db;
+    docs = [];
+    const docsArray: Array<Doc[]> = await Promise.all(
+      channelsData.channels.map(async (channel, channelIndex) => {
+        const result = await parser.parseURL(channel.url);
+        return (
+          result?.items.map((item) => ({
+            ...item,
+            author: channel,
+            authorId: channelIndex
+          })) ?? []
+        );
+      })
+    );
+    const maxHeap = new MaxHeap<HeapNode>((node1, node2) => {
+      let dateNode1 = new Date(node1[0].pubDate as string);
+      let dateNode2 = new Date(node2[0].pubDate as string);
+      if (dateNode1 > dateNode2) {
+        return 1;
+      }
+      if (dateNode1 == dateNode2) {
+        return 0;
+      }
+      return -1;
     });
+    for (let i = 0; i < docsArray.length; i++) {
+      if (docsArray[i].length > 0) {
+        maxHeap.push([docsArray[i][0], i, 0]);
+      }
+    }
+    while (maxHeap.length > 0) {
+      const pop = maxHeap.pop() as HeapNode;
+      docs.push(pop[0]);
+      if (pop[2] !== docsArray[pop[1]].length - 1) {
+        maxHeap.push([docsArray[pop[1]][pop[2] + 1], pop[1], pop[2] + 1]);
```

Comment:
Could you make another change to the `HeapNode` type so it has named fields, please?

This is where we need it, reading this line of code won't provide much information about what are those `pop[1]` `pop[2]` is. I think it will be hard to maintain in the future.

Reply:
Or maybe you can use destructuring to give a more meaningful name on line 61, like:

```
const [doc, whatever1, whatever2] = maxHeap.pop() as HeapNode;
```

Reply:
Yeah I have thought about it too, but I wasn't really sure how to properly naming these variables. Please review my newest changes, does it make sense now? Thank you!

Reply:
Perfect! It looks more clear now. Thank you so much!